MobileNav/ViewControllers/Map/MapBoxViewController.swift

```diff
@@ -0,0 +1,286 @@
+        super.viewDidLoad()
+        setup()
+    }
+
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        NotificationCenter.default.post(name: SDLDidUpdateProjectionView, object: nil)
+    }
+}
+
+// MARK: - Button Actions
+
+extension MapBoxViewController {
+    @IBAction func searchButtonTapped(_ sender: UIButton) { performSearch() }
+    @IBAction func zoomInButtonTapped(_ sender: UIButton) { mapManager.zoomIn() }
+    @IBAction func zoomOutButtonTapped(_ sender: UIButton) { mapManager.zoomOut() }
+    @IBAction func menuButtonTapped(_ sender: UIButton) { presentSettings() }
+    @IBAction func centerLocationButtonTapped(_ sender: UIButton) {
+        if let userLocation = userLocation {
+            mapManager.centerLocation(lat: userLocation.coordinate.latitude, long: userLocation.coordinate.longitude)
+        } else {
+            let alert = UIAlertController(title: "Cant find user location", message: "Make sure you have location permissions enabled", preferredStyle: .alert)
+            let action = UIAlertAction(title: "OK", style: .default, handler: nil)
+            alert.addAction(action)
+        }
+    }
+}
+
+// MARK: - Helper Functions
+
+extension MapBoxViewController {
+
+    func setup() {
+        DispatchQueue.main.async {
+            self.setupObservers()
+            self.setupTouchManager()
+            self.setupButtons()
+            self.setupUserLocation()
+        }
+    }
+
+    private func setupObservers() {
+        NotificationCenter.default.addObserver(self, selector: #selector(presentOffScreen), name: .offScreenConnected, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(setupUserLocation), name: .locationUpdated, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(hideSubscribedButtons), name: .hideSubscribedButtons, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(showHiddenButtons), name: .showHiddenButtons, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(centerMapOnLocation), name: .centerMapOnPlace, object: nil)
+    }
+
+    private func setupTouchManager() {
+        mapTouchHandler = mapManager.mapManagerTouchHandler
+        menuTouchHandler = menuButton.buttonTouchHandler
+        ProxyManager.sharedManager.sdlManager.streamManager?.touchManager.touchEventDelegate = self
+    }
+
+    private func setupButtons() {
+        searchButton.setImage(UIImage(named: "search"), for: .normal)
+        centerMapButton.setImage(UIImage(named: "center"), for: .normal)
+        zoomInButton.setImage(UIImage(named: "zoom_in"), for: .normal)
+        zoomOutButton.setImage(UIImage(named: "zoom_out"), for: .normal)
+
+        // Hide buttons if we are going to subscribe them
+        if let rpcVersion = ProxyManager.sharedManager.rpcVersion {
+            if rpcVersion >= 6 {
+                hideSubscribedButtons()
```

Comment:
There is no guarantee that modules supporting RPC v6.0+ will support the navigation subscribe buttons so we need to also check the button capabilities to see if the button is supported: 

```
let navZoomInSubscribeButtonSupported = ProxyManager.sharedManager.sdlManager.systemCapabilityManager.defaultMainWindowCapability?.buttonCapabilities?.contains(where: { $0.name == SDLButtonName.navZoomIn })
```